PatzHum/Matcha
# Developed base webapp syncing with firebase webservice

### What does this accomplish?
Implements basic syncing of the editor textbox with firebase backend

### What is the scope of this change?
The editor tab of the webapp

### How can I test this?
Open the webapp -> open the editor tab -> sign in -> type something in the editor box -> click `Update` -> open firebase database viewer -> find your user id -> the typed data should be visible

- [x] This build compiles
- [x] I have tested this and AFAIK these changes do not add any additional crashes/bugs
